lib/util/cache/repository/impl/client/cache-client-factory.ts

```diff
@@ -0,0 +1,36 @@
+import { GlobalConfig } from '../../../../../config/global';
+import type { RepositoryCacheType } from '../../../../../config/types';
+import { logger } from '../../../../../logger';
+import type { CacheClient } from '../../types';
+import { LocalRepoCache } from './local';
+
+export class CacheClientFactory {
+  private static client: CacheClient | null;
+
+  static get(repository: string, cacheType: RepositoryCacheType): CacheClient {
+    if (this.client) {
+      return this.client;
+    }
+
+    const platform = GlobalConfig.get('platform');
+
+    switch (cacheType) {
+      case 'local':
+        this.client = new LocalRepoCache(platform!, repository);
+        break;
+      // istanbul ignore next: untestable
+      default:
+        this.client = new LocalRepoCache(platform!, repository);
+        logger.info(
+          `Repository cache type not supported using type "local" instead`
+        );
```

Comment:
```suggestion
        logger.warn(
          { cacheType },
          `Repository cache type not supported using type "local" instead`
        );
```

Reply:
i dont think any change will be required. i run locally without any changes and repo cache works like it should. 
repoCacheTypes defaults to 'local'

Reply:
ive also run comparison between logs from branch main run vs logs from this feature branch run (same config same repo).
logs are basically the same other than timestamps that is.

thats i guess the most important log message in this context
![image](https://user-images.githubusercontent.com/97394622/184241924-50cf3e69-240c-46c6-8ae6-80005c3bbb90.png)